### ISSUE TITLE
Number of bases not enforced when tax_abundance is set?

### ISSUE THREAD

**Issue by baptwr** [metagenlab/MeSS#28]
Hello,

Thank you for your tool. I have encountered something I don't understand. However I am not sure if it is a bug or if it is wanted.

**MeSS version**

v0.9.1. Built from source.

**Describe the bug**

When using an input table with the `tax_abundance` value set and the `--bases` argument of `mess simulate`, the produced samples have bigger sequencing depth than set with `--bases`.

**Minimal example**

Use the following input table using [example genomes](https://github.com/metagenlab/MeSS/tree/main/mess/test_data/fastas):

```
fasta	tax_id	tax_abundance	sample
GCF_000418345.1	1280	0.2	sample1
GCF_003812505.1	1290	0.8	sample1
GCF_000418345.1	1280	0.1	sample2
GCF_003812505.1	1290	0.9	sample2
```

Run the following `mess simulate` command:

```bash
mess simulate --input test_table.tsv --fasta mess/test_data/fastas --threads 10 --bases 2G
```
Check simulated samples stats:

```bash
seqkit stat mess_out/fastq/*
```

|file                           |format|type|num_seqs|sum_len   |min_len|avg_len|max_len|
|:------------------------------|:-----|:---|:-------|:---------|:------|:------|:------|
|mess_out/fastq/sample1_R1.fq.gz|FASTQ |DNA |24926434|3738965100|150    |150.0  |150    |
|mess_out/fastq/sample1_R2.fq.gz|FASTQ |DNA |24926434|3738965100|150    |150.0  |150    |
|mess_out/fastq/sample2_R1.fq.gz|FASTQ |DNA |24173018|3625952700|150    |150.0  |150    |
|mess_out/fastq/sample2_R2.fq.gz|FASTQ |DNA |24173018|3625952700|150    |150.0  |150    |

For Nanopore:

```bash
mess simulate --input test_table.tsv --fasta mess/test_data/fastas --threads 10 --bases 2G --tech nanopore
seqkit stat mess_out/fastq/*
```

|file                        |format|type|num_seqs|sum_len   |min_len|avg_len|max_len|
|:---------------------------|:-----|:---|:-------|:---------|:------|:------|:------|
|mess_out/fastq/sample1.fq.gz|FASTQ |DNA |837930  |7478571786|95     |8925.1 |75727  |
|mess_out/fastq/sample2.fq.gz|FASTQ |DNA |810676  |7252585866|93     |8946.3 |75844  |

Samples don't have 2 Gbp.

Is it normal behavior since `--bases` is described as "per sample base counts" ? If so, what is the correct way to simulate samples with taxa abundances and sequencing depth both set?

Thank you in advance.

**Comment by farchaab** [comment]
Hello @baptwr and thank you for using `mess` !

We realized with @CarraraAlessia  that tax_abundance calculations were wrong and we fixed them in the [dev branch ](https://github.com/metagenlab/MeSS/tree/dev) (fixed with a59196d2d993a9f79ec6ef8fb59892ed3315d318 and 2dbd7f36095a70fcff4bdfe8d3338588e45e2d90)

Can you try to checkout the dev branch and update us if it fixed your bug ?

The changes will be included in the next release.

**Comment by farchaab** [comment]
Hello @baptwr, 

I did some more digging and it turns out that the bug is due to concatenating fastqs several times. This happens when the input contains more than one sample with the same genomes. Which is the case in the example below:

```
fasta	tax_id	tax_abundance	sample
GCF_000418345.1	1280	0.2	sample1
GCF_003812505.1	1290	0.8	sample1
GCF_000418345.1	1280	0.1	sample2
GCF_003812505.1	1290	0.9	sample2
```

I fixed the issue with 61ab8592d383889d647a71911e72fc11ace2155e in the `dev` branch, and reran `mess` for both Illumina and Nanopore :

* Illumina
```sh
mess simulate --input test_table.tsv --fasta fastas --threads 17 --bases 2G -o mess_out/illumina --sdm apptainer
seqkit stats mess_out/illumina/fastq/*
```
|file                                    |format|type|num_seqs|sum_len  |min_len|avg_len|max_len|
|:---------------------------------------|:-----|:---|:-------|:--------|:------|:------|:------|
|mess_out/illumina/fastq/sample1_R1.fq.gz|FASTQ |DNA |6666096 |999914400|150    |150.0  |150    |
|mess_out/illumina/fastq/sample1_R2.fq.gz|FASTQ |DNA |6666096 |999914400|150    |150.0  |150    |
|mess_out/illumina/fastq/sample2_R1.fq.gz|FASTQ |DNA |6666041 |999906150|150    |150.0  |150    |
|mess_out/illumina/fastq/sample2_R2.fq.gz|FASTQ |DNA |6666041 |999906150|150    |150.0  |150    |

* Nanopore
```sh
mess simulate --input test_table.tsv --fasta fastas --threads 17 --bases 2G -o mess_out/nanopore --sdm apptainer --tech nanopore
seqkit stats mess_out/nanopore/fastq/*
```

|file                                 |format|type|num_seqs|sum_len   |min_len|avg_len|max_len|
|:------------------------------------|:-----|:---|:-------|:---------|:------|:------|:------|
|mess_out/nanopore/fastq/sample1.fq.gz|FASTQ |DNA |223587  |2000000361|95     |8945.1 |70023  |
|mess_out/nanopore/fastq/sample2.fq.gz|FASTQ |DNA |223825  |2000000279|94     |8935.6 |75844  |

I will include the fix in the next release.

Let me know if you find any other issues !
